### DISCUSSION TITLE
Unable to download diagram into PDF - "convert.diagrams.net didn’t send any data"

### DISCUSSION THREAD

**Discussion by jkuanec** [jgraph/drawio#2540] [Q&A]
Hello,

I have been using diagrams.net for over a year now and have downloaded my diagrams into PDFs many times. Recently, I have not successfully downloaded my diagrams into PDFs (other formats work). If I share my diagram with my partner and ask him to download it as PDF on his PC, it works without any issue. 

Can anyone shed any light as to what is potentially the problem? 

ps. I cannot download the desktop version as I am working on a work laptop and the desktop version of diagrams.net is not yet approved for installation. 

Thank you,

Jenny

![image](https://user-images.githubusercontent.com/97551300/149006954-c60021e7-6175-477e-86ca-01709a9e966a.png)

**Comment by davidjgraph** [comment]
We'll take a look at the logs, but printing to PDF will produce the same output.

**Reply by jkuanec** [reply]
Hi,

Thank you for taking the time to reply! While the print to PDF option does work, the margins are significantly wider relative to exporting directly as a PDF:

Print to PDF:
![image](https://user-images.githubusercontent.com/97551300/149179250-c42f9043-0b0f-4e95-8b41-f7be32a3a071.png)

Export as PDF:
![image](https://user-images.githubusercontent.com/97551300/149179632-7ec74088-18fb-472f-b9db-d87aa26d1854.png)


The difference is minor but I much prefer the aesthetic of exporting it as a PDF.

Thank you!

**Comment by m-mohamedin** [comment]
Is the issue affecting all your diagrams or only one?

**Reply by jkuanec** [reply]
It's affecting the new diagrams I'm creating. I just downloaded one that I previously made (about a year ago) and it downloaded without any issue as a PDF

**Reply by m-mohamedin** [reply]
Is it possible to share the diagram with us? You can send it to my email mohamedin@draw.io if you cannot post it here

**Reply by jkuanec** [reply]
For the specific diagram in question, I needed to delete some of the images/description as it is data that cannot be openly shared. Upon doing that, I downloaded the diagram again (just to test it out) and it worked! I re-downloaded the original file (with all the images and description) and it worked surprisingly. 

I went and tried to download other diagrams that I was previously unsuccessful in downloading and it sort of worked. I am unable to download the diagrams that have many elements to it though:
![image](https://user-images.githubusercontent.com/97551300/149199378-94b9107c-59c5-4465-af26-5287e1f4f020.png)

The diagram with the screenshot "Error!" has over 30 pages in total with text and images - I wasn't able to download it this time around but was able to download the "Report" / "Page-1" tab with 35 pages. My partner has successfully downloaded the complete file though without a hitch. 

![image](https://user-images.githubusercontent.com/97551300/149199843-6733e061-7830-4583-9f36-3baca9a29814.png)


Is it possible that there is a memory issue for downloading large diagrams?

**Reply by m-mohamedin** [reply]
Sometimes the server times out if the diagram is taking long time to process. Without the actual file we cannot find out the exact reason.

**Reply by jkuanec** [reply]
I can send you another diagram that I created recently (also not working now).

**Comment by davidjgraph** [comment]
Could you please test another browser? Do you have a different network you can test with the browser that failed?

**Reply by jkuanec** [reply]
I just tried on Mozilla and I am still unable to export my most recent diagram as a PDF.

**Reply by jkuanec** [reply]
Here is the error message from Mozilla finally:

![image](https://user-images.githubusercontent.com/97551300/149197407-b9d5091d-ddda-4b0a-88f5-5adbc7ac7489.png)
